src/flint.h

```diff
@@ -438,14 +451,20 @@ mp_limb_t FLINT_BIT_COUNT(mp_limb_t x)
 FLINT_DLL int parse_fmt(int * floating, const char * fmt);
 
 FLINT_DLL int flint_printf(const char * str, ...); /* flint version of printf */
-FLINT_DLL int flint_vprintf(const char * str, va_list ap); /* va_list version of flint_printf */
-FLINT_DLL int flint_fprintf(FILE * f, const char * str, ...); /* flint version of fprintf */
 FLINT_DLL int flint_sprintf(char * s, const char * str, ...); /* flint version of sprintf */
 
 FLINT_DLL int flint_scanf(const char * str, ...); /* flint version of scanf */
-FLINT_DLL int flint_fscanf(FILE * f, const char * str, ...); /* flint version of fscanf */
 FLINT_DLL int flint_sscanf(const char * s, const char * str, ...); /* flint version of sscanf */
 
+#ifdef FLINT_HAVE_VA_LIST
+FLINT_DLL int flint_vprintf(const char * str, va_list ap); /* va_list version of flint_printf */
+#endif
+
+#ifdef FLINT_HAVE_FILE
+FLINT_DLL int flint_fprintf(FILE * f, const char * str, ...); /* flint version of fprintf */
```

Comment:
Have you considered just doing `typedef void * FLINT_FILE` somewhere and using that in those headers?

Reply:
Sorry for my late answer. So I do not like replacing `FILE` just like that everywhere. If you have a filestream you want to use, you have obviously included `stdio.h`. However, for other types where you typically initialize a struct with a filestream, such as stream objects in `gr.h`, this is actually a great idea. I am first going to do a PR where I deinline printer functions, and then we can argue whether we should include `stdio.h` by default or not ;-)